lib/rbi/rewriters/filter_versions.rb

```diff
@@ -0,0 +1,113 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RBI
+  module Rewriters
+    # Take a gem version and filter out all RBI that is not relevant to that version based on @version annotations
+    # in comments. As an example:
+    #
+    # ~~~rb
+    # tree = Parser.parse_string(<<~RBI)
+    #   class Foo
+    #     # @version > 0.3.0
+    #     def bar
+    #     end
+    #
+    #     # @version <= 0.3.0
+    #     def bar(arg1)
+    #     end
+    #   end
+    # RBI
+    #
+    # Rewriters::FilterVersions.filter(tree, Gem::Version.new("0.3.1"))
```

Comment:
I think we should provide a `Tree#filter_versions!` method like we do on other rewriter (see the sorting for example: https://github.com/Shopify/rbi/blob/main/lib/rbi/rewriters/sort_nodes.rb#L115-L123).